src/Microsoft.Health.Fhir.SqlServer/Features/Storage/SqlServerTaskConsumer.cs

```diff
@@ -157,13 +161,16 @@ public async Task<TaskInfo> GetNextMessagesAsync(int taskHeartbeatTimeoutThresho
                         Result = result,
                     };
                 }
-            }
-            catch (SqlException e) when (e.Number == 2812)
-            {
-                _logger.LogWarning(e, "Schema is not initialized - {ex.Message}", e.Message);
-            }
+                catch (SqlException sqlEx)
+                {
+                    if (sqlEx.Number == SqlErrorCodes.NotFound)
```

Comment:
>if (sqlEx.Number == SqlErrorCodes.NotFound) [](http://example.com/codeflow?start=20&length=43)

I would suspect it would not return notfound, which case would return notfound error?

Reply:
I used the catch code from other tasks. I am assuming it catches exception when a resource doesn't exist. Also, if a particular schema version doesn't have the SP

Reply:
I am still wondering if we need an explicit check for NotFound error here since in all other methods(CompleteAsync() or KeepAliveAsync()), I assume since we are reading the task with specific taskId and runId and if we could not find the task then we are throwing the TaskNotExistException but here we are just checking if the next message exists in the queue so probably we don't need a catch here. @tongwu-msft Please correct me if I have misunderstood anything here